components/TypingAnimation.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react';
+import { Box } from '@mui/joy';
+
+interface TypingProps {
+  text: string;
+  delay: number;
+  infinite?: boolean;
+  pause?: number; // Optional delay before restarting
+}
+
+export default function TypingAnimation({ text, delay, infinite, pause }: TypingProps) {
+  const [currentText, setCurrentText] = useState('');
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [isPaused, setIsPaused] = useState(false);
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+
+    if (isPaused) {
+      timeout = setTimeout(() => {
+        setIsPaused(false);
+        setCurrentIndex(0);
+        setCurrentText('');
+      }, pause);
+    } else if (currentIndex < text.length) {
+      timeout = setTimeout(() => {
+        setCurrentText((prevText) => prevText + text[currentIndex]);
+        setCurrentIndex((prevIndex) => prevIndex + 1);
+      }, delay);
+    } else if (infinite) {
+      setIsPaused(true);
+    }
+
+    return () => clearTimeout(timeout);
+  }, [currentIndex, delay, infinite, text, isPaused, pause]);
+
+  return <Box height={100}>{currentText}</Box>;
```

Comment:
You'll just need to remove the height={100} so that the text will sit in the center again. Right now it's like this:

<img width="387" alt="Screen Shot 2024-09-05 at 3 52 32 pm" src="https://github.com/user-attachments/assets/6377229a-1cb0-4411-a322-320061747ba6">